modules/simulation/src/main/resources/schemas/neurosciencegraph/simulation/bluepyefeconfiguration/v0.1.2.json

```diff
@@ -0,0 +1,113 @@
+{
+  "@context": [
+    "{{base}}/contexts/neurosciencegraph/core/schema/v0.1.0",
+    {
+      "this": "{{base}}/schemas/neurosciencegraph/simulation/bluepyefeconfiguration/v0.1.2/shapes/"
+    }
+  ],
+  "@type": "nxv:Schema",
+  "imports": [
+    "{{base}}/schemas/neurosciencegraph/commons/entity/v0.1.0",
+    "{{base}}/schemas/neurosciencegraph/commons/typedlabeledontologyterm/v0.1.2"
+  ],
+  "shapes": [
+    {
+      "@id": "this:BluePyEfeConfigurationShape",
+      "@type": "sh:NodeShape",
+      "label": "BluePyEfe configuration shape",
+      "comment": "BluePyEfe features extraction configuration shape.",
+      "targetClass": "nsg:BluePyEfeConfiguration",
```

Comment:
I'll try may be to explain how I see the usage of type and shape. 

It is perfectly valid to have type and schema name aligned just like you did. The concern here is more about style rather than correctness.

It is not always necessary to have the same name for type and schema.

If the type name always follow the schema name then you end up with a long list of very specific types. I agree that those types are very easy to query (you just filter entities by type nsg:BluePyEfeConfiguration if you want to get config files for BluePyEfe) but they are very hard to manage in long term.

Here is how I recommend to handle this type of situation:
 * A single type (e.g. nsg:Configuration) can have multiple shapes associated (through targetClass) (e.g. simulation/bluepyefeconfiguration, ...). 
 * to get config files for BluePyEfe, you can get entities of type nsg:Configuration and conformant (nxv:schema) to the schema simulation/bluepyefeconfiguration .

Reply:
Will it change in v1 how we query data? Will org/domain remain as a possible collection filter?

Reply:
Also as far as I understand shapes for payload validation are selected based on targetClass, so if we have one type nsg:Configuration provided in payload all the shapes with the same targetClass will have to be validated, correct? And all different configuration payloads are different, so I'm not getting how they can pass validation against multiple different shapes...

Reply:
org/domain will become org/project so almost the same thing in Nexus v1. It is always possible to filter data by type and schema independantly of the Nexus version (v0, v1).

Validation is triggered when you submit one entity against one schema which potentially involve imports other schemas. Among the shapes in a schema import closure, there should be **exactly one shape with a target declaration (targetClass, targetObjectsO, targetSubjectsOf or targetrNode)**. This is how we design the schemas and not something Nexus enforced . It is the submitted entity or a part of it that is being selected for validation through target declaration in a given shape, not the shapes.
So to precisely answer your question:

> Also as far as I understand shapes for payload validation are selected based on targetClass, so if we have one type nsg:Configuration provided in payload all the shapes with the same targetClass will have to be validated, correct?

No, this situation will never occur.

Reply:
Ok, thanks for the explanation.
Another question, hypothetical one :) how can I query all BluePyEfeConfigurations independent of the organizations? Can I just set filter to: "op":"eq", "path":"nxv:schema", "value":"simulation/bluepyefeconfiguration"?